packages/vc-export/src/types.ts

```diff
@@ -5,82 +5,133 @@
-  KILT_SELF_SIGNED_PROOF_TYPE,
+  KILT_ATTESTER_DELEGATION_V1_TYPE,
+  KILT_ATTESTER_LEGITIMATION_V1_TYPE,
+  KILT_CREDENTIAL_IRI_PREFIX,
+  KILT_REVOCATION_STATUS_V1_TYPE,
+  W3C_CREDENTIAL_CONTEXT_URL,
+  W3C_PRESENTATION_TYPE,
 } from './constants.js'
 
-export interface Proof {
-  type: string
-  created?: string
-  proofPurpose?: string
-  [key: string]: any
-}
-
 export type IPublicKeyRecord = ConformingDidKey
 
-export interface SelfSignedProof extends Proof {
-  type: typeof KILT_SELF_SIGNED_PROOF_TYPE
-  verificationMethod: IPublicKeyRecord['id'] | IPublicKeyRecord
-  signature: string
-  challenge?: string
-}
-export interface AttestedProof extends Proof {
-  type: typeof KILT_ATTESTED_PROOF_TYPE
-  attester: string
-}
-export interface CredentialDigestProof extends Proof {
-  type: typeof KILT_CREDENTIAL_DIGEST_PROOF_TYPE
-  // map of unsalted property digests and nonces
-  nonces: Record<string, string>
-  // salted hashes of statements in credentialSubject to allow selective disclosure.
-  claimHashes: string[]
+export interface KiltAttestationProofV1 {
+  type: typeof ATTESTATION_PROOF_V1_TYPE
+  block: string
+  commitments: string[]
+  salt: string[]
 }
 
-export interface CredentialSchema {
-  '@id': string
-  '@type': typeof JSON_SCHEMA_TYPE
-  schema: ICType
+export interface JsonSchemaValidator2018 {
+  id: string
+  type: typeof JSON_SCHEMA_TYPE
+  schema?: ICType
   modelVersion?: string
   name?: string
   author?: string
   authored?: string
-  proof?: Proof
+  // proof?: Proof
 }
 
-export interface VerifiableCredential {
-  '@context': [typeof DEFAULT_VERIFIABLECREDENTIAL_CONTEXT, ...string[]]
-  // the credential types, which declare what data to expect in the credential
-  type: [typeof DEFAULT_VERIFIABLECREDENTIAL_TYPE, ...string[]]
+export interface KiltRevocationStatusV1 {
+  id: Caip2ChainId
+  type: typeof KILT_REVOCATION_STATUS_V1_TYPE
+}
+
+export interface CredentialSubject {
+  '@context': {
+    '@vocab': string
+  }
+  id: DidUri
+  [k: string]: Record<string, unknown> | string | number | boolean
+}
+
+interface IssuerBacking {
   id: string
-  // claims about the subjects of the credential
-  credentialSubject: Record<string, AnyJson>
-  // the entity that issued the credential
-  issuer: string
-  // when the credential was issued
+  type: string
+}
+
+export interface KiltAttesterLegitimationV1 extends IssuerBacking {
+  id: VerifiableCredential['id']
+  type: typeof KILT_ATTESTER_LEGITIMATION_V1_TYPE
+  verifiableCredential?: VerifiableCredential
+}
+
+export interface KiltAttesterDelegationV1 extends IssuerBacking {
+  id: `kilt:delegation/${string}`
+  type: typeof KILT_ATTESTER_DELEGATION_V1_TYPE
+  delegators?: DidUri[]
+}
+
+export interface VerifiableCredential {
+  /**
+   * References to json-ld contexts defining the terms used.
+   */
+  '@context': typeof DEFAULT_CREDENTIAL_CONTEXTS
+  /**
+   * Credential identifier.
+   */
+  id: `${typeof KILT_CREDENTIAL_IRI_PREFIX}${string}`
```

Comment:
The idea is to fill this repo (https://github.com/ChainAgnostic/namespaces/tree/main) with a polkadot-specific section about chain IDs (this will most likely not change from what we have here), and assets. There was some discussion about having a namespace for generic on-chain attestations, that is why I am suggesting to not simply use `kilt` as the namespace, but rather use some KILT-derived references for this on-chain attestation namespace. For CTypes the situation is a bit different, but I guess we could define them as on-chain credential schema definitions, and define a namespace for it as well. The references would then define instructions on how to interpret the results, namely what to do with an attestation hash or a CType ID. This is all wishy-washy still, so I am not sure what's the best way to proceed here for stuff like credentials that can be persisted and can be around potentially forever unless we deprecate them.

Reply:
> Didn't we decide in favor of using CAIP-19 identifiers for credentials, CTypes,and everything else? This would make it possible to issue public credentials to these credentials, as with the Kiltagram example. Something like `<kilt_caip_2>:att:kilt-attestation:<attestation_hash>`.

This is not in the specs right now, and I think for a good reason: credentials are not on-chain artifacts, attestations are. I use a CAIP identifier for attestations in the specs, but not for credentials; and I think doing so would be misleading. Because the data structure that is being identified simply is not on chain.
While I wouldn't want to use a CAIP id here, I'm happy to switch to a different uri pattern, as I'm not to convinced about this one. Suggestions welcome.

> Now that I think of it, it would make sense to also change what the CType ID is, since we have a breaking change we will roll out soon anyway. That would be future-proof and avoid breaking again. Then we can start issuing credentials to CTypes as well.

In theory we can change the CType id format any time, unfortunately the current ids are hashed into existing credentials though. Using a different id scheme for CTypes adhering to the meta schema would of course be possible, but it's not on the roadmap right now - and it would make the transition from the legacy meta schema to the new one way less smooth.
We can still refer to CTypes by their CAIP identifier though if we want to issue credentials to them, I see no issue with that.

Reply:
Would be cool though if we could properly define CAIP namespaces for attestations and delegations before this goes live, so if there is a consensus forming let me know! I was also thinking that pallet and storage names could be obvious choices for asset namespace and class, but the restriction to 8 characters for the namespace makes that a little tricky. Maybe truncating helps? Anyway, these discussions should probably be had on the [specs repo](https://github.com/KILTprotocol/spec-KiltCredentialV1).

Reply:
> This is not in the specs right now, and I think for a good reason: credentials are not on-chain artifacts, attestations are. I use a CAIP identifier for attestations in the specs, but not for credentials; and I think doing so would be misleading.

Right, probably it would be wrong to use the `att` namespace, but we could perhaps use the `cred` namespace to identify generic verifiable credentials, in which we would fit current KILT credentials, new VC-compatible credentials, and public credentials we well (?)

> We can still refer to CTypes by their CAIP identifier though if we want to issue credentials to them, I see no issue with that.

That sounds like a good compromise to me!

Reply:
> Right, probably it would be wrong to use the `att` namespace, but we could perhaps use the `cred` namespace to identify generic verifiable credentials, in which we would fit current KILT credentials, new VC-compatible credentials, and public credentials we well (?)

Mhh as I said it feels inappropriate to me to use a CAIP at all if we're trying to identify off-chain credentials. For public credentials it kind of makes sense to me (even though strictly speaking, as is also the case with CTypes, you seem to be identifying primarily the "pointer" to the actual data of which the credential is composed, because the latter is not in state). But I'd resist trying to CAIPify stuff that does not live on a blockchain.

Reply:
> Mhh as I said it feels inappropriate to me to use a CAIP at all if we're trying to identify off-chain credentials.

I see. Perhaps you are misguided by your assumption of what a CAIP should do. But in theory it's Chain Agnostic, meaning that it could also be used for stuff that does not necessarily live on a blockchain. But this is still an open point I am also discussing with others, so we can let it be for now.

> even though strictly speaking, as is also the case with CTypes, you seem to be identifying primarily the "pointer" to the actual data of which the credential is composed, because the latter is not in state

An asset could be defined as a combination of on-chain and off-chain information, hence I would think it's totally viable to define a CType asset as a mix of on-chain and off-chain information, although off-chain is not REALLY offchain as can be an IPFS link, but rather still conceptually quite tied to the blockchain itself.